test/e2e/node/pod_resize.go

```diff
@@ -768,6 +769,201 @@ func doPodResizeRetryDeferredTests(f *framework.Framework) {
+		//	      - pod3's resize should make room for pod4's resize.
+		//     6. Delete pod1, verify the chain of deferred resizes is actuated.
+
+		podClient := e2epod.NewPodClient(f)
+		nodes, err := e2enode.GetReadySchedulableNodes(ctx, f.ClientSet)
+		framework.ExpectNoError(err, "failed to get running nodes")
+		gomega.Expect(nodes.Items).ShouldNot(gomega.BeEmpty())
+		framework.Logf("Found %d schedulable nodes", len(nodes.Items))
+
+		ginkgo.By("Find node CPU and memory resources available for allocation!")
+		node := nodes.Items[0]
+
+		nodeAllocatableMilliCPU, initNodeAvailableMilliCPU := getNodeAllocatableAndAvailableMilliCPUValues(ctx, f, &node)
+		framework.Logf("Node '%s': NodeAllocatable MilliCPUs = %dm. MilliCPUs currently available to allocate = %dm.",
+			node.Name, nodeAllocatableMilliCPU, initNodeAvailableMilliCPU)
+
+		nodeAllocatableMem, initNodeAvailableMem := getNodeAllocatableAndAvailableMemoryValues(ctx, f, &node)
+		framework.Logf("Node '%s': NodeAllocatable Memory = %d. Memory currently available to allocate = %d.",
+			node.Name, nodeAllocatableMem, initNodeAvailableMem)
+
+		initialCPUQuantity := resource.NewMilliQuantity(initNodeAvailableMilliCPU/4, resource.DecimalSI)
+		initialMemoryQuantity := resource.NewQuantity(initNodeAvailableMem/5, resource.DecimalSI)
+		framework.Logf("initial CPU request is '%dm'", initialCPUQuantity.MilliValue())
+		framework.Logf("initial Memory request is '%d'", initialMemoryQuantity.Value())
+
+		c := []podresize.ResizableContainerInfo{
+			{
+				Name: "c",
+				Resources: &cgroups.ContainerResources{
+					CPUReq: initialCPUQuantity.String(),
+					MemReq: initialMemoryQuantity.String(),
+				},
+			},
+		}
+
+		tStamp := strconv.Itoa(time.Now().Nanosecond())
+		testPod1 := podresize.MakePodWithResizableContainers(f.Namespace.Name, "testpod1", tStamp, c)
+		testPod1 = e2epod.MustMixinRestrictedPodSecurity(testPod1)
+		e2epod.SetNodeAffinity(&testPod1.Spec, node.Name)
+
+		testPod2 := podresize.MakePodWithResizableContainers(f.Namespace.Name, "testpod2", tStamp, c)
+		testPod2 = e2epod.MustMixinRestrictedPodSecurity(testPod2)
+		e2epod.SetNodeAffinity(&testPod2.Spec, node.Name)
+
+		testPod3 := podresize.MakePodWithResizableContainers(f.Namespace.Name, "testpod3", tStamp, c)
+		testPod3 = e2epod.MustMixinRestrictedPodSecurity(testPod3)
+		e2epod.SetNodeAffinity(&testPod3.Spec, node.Name)
+
+		testPod4 := podresize.MakePodWithResizableContainers(f.Namespace.Name, "testpod4", tStamp, c)
+		testPod4 = e2epod.MustMixinRestrictedPodSecurity(testPod4)
+		e2epod.SetNodeAffinity(&testPod4.Spec, node.Name)
+
+		testPods := []*v1.Pod{testPod1, testPod2, testPod3, testPod4}
+
+		for i, pod := range testPods {
+			ginkgo.By(fmt.Sprintf("Create pod '%s' that fits the node '%s'", pod.Name, node.Name))
+			testPods[i] = podClient.CreateSync(ctx, pod)
+			gomega.Expect(testPods[i].Status.Phase).To(gomega.Equal(v1.PodRunning))
+			gomega.Expect(testPods[i].Generation).To(gomega.BeEquivalentTo(1))
+		}
```

Comment:
I think you can just use `CreateBatch` for this? It's basically doing this in parallel. The running check is redundant with CreateSync (which waits for the pods to be running). If the generation check is important, you'll need to verify it separately.